lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -71,8 +76,21 @@ def relationship_value_for(association, options)
         return association.options[:virtual_value] if association.options[:virtual_value]
         return unless association.serializer && association.serializer.object
 
+        if options[:fields].is_a? Array
```

Comment:
this snippet of code is somewhat problematic.. I believe what it says is

1. If options fields is a collection of class Array (bug: should checking if `respond_to?(:to_ary)`)
1. collect all elements of class Hash  (or responds to keys is probably fine)
1. If the collection responds to first, call first (bug: it will always respond to first. and if we only want the first item in the collection, we should just use detect or better lazy and take)
1. If the previous step's result responds to `:[]`, i.e. the collection had a hash in it, get the value keyed by the association name

I think this could be rewritten as

```ruby
fields = Array(options[:fields]).lazy.select {|elem| elem.respond_to?(:keys) }.take(1).map {|elem| association.name }.force.compact.first
```

Would be nice for this to be tested separately or have a descriptive name so it's easier to follow what it's doing, as well

:)